frontend/src/metabase/query_builder/components/EditableText/EditableText.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from "react";
+
+import { Root, BorderedInput } from "./EditableText.styled";
+
+import { KEY_ESCAPE } from "metabase/lib/keyboard";
+
+interface Props {
+  value: string;
+  onChange?: (val: string) => void;
+}
+
+export const EditableText = ({ value, onChange }: Props) => {
+  const [memo, setMemo] = useState(value);
+
+  const handleChange = (e: React.ChangeEvent) => {
+    const target = e.target as HTMLInputElement;
+    setMemo(target.value);
+  };
+
+  const handleBlur = (e: React.ChangeEvent) => {
+    if (onChange) {
+      const target = e.target as HTMLInputElement;
+      onChange(target.value);
+    }
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
+    if (e.key === KEY_ESCAPE) {
+      setMemo(value);
+    }
+  };
+
+  return (
+    <Root data-replicated-value={memo}>
+      <BorderedInput
```

Comment:
Should the component name reflect what it renders? e.g.
```suggestion
      <BorderedTextArea
```